File: src/cxx/rust/cxx_format.rs
#[cxx::bridge(namespace = "com::keygenqt::aurora_rs_mcp")]
mod ffi {
    unsafe extern "C++" {
        include!("aurora-rs-mcp/src/cxx/cpp/format/cxx_format.h");

        type CxxFormat;

        fn new_format() -> UniquePtr<CxxFormat>;
        fn time(self: Pin<&mut CxxFormat>, data: String) -> String;
    }
}

/// Example execute
pub fn time(data: &str) -> String {
    ffi::new_format().pin_mut().time(data.to_string())
}

File: src/cxx/cpp/format/cxx_format.cpp
#include "aurora-rs-mcp/src/cxx/cpp/format/cxx_format.h"
#include "aurora-rs-mcp/src/cxx/rust/cxx_format.rs.h"

#include <ctime>
#include <cstring>

namespace ru
{
  namespace kotdath
  {
    namespace aurora_rs_mcp
    {
      std::unique_ptr<CxxFormat> new_format()
      {
        return std::make_unique<CxxFormat>();
      }

      CxxFormat::CxxFormat() {}

      rust::String CxxFormat::time(rust::String format)
      {
        std::time_t time = std::time({});
        char formatData[format.size()];
        strcpy(formatData, format.c_str());
        std::strftime(formatData, format.size(), "%FT%TZ", std::gmtime(&time));
        return formatData;
      }
    } // namespace aurora_rs_mcp
  } // namespace kotdath
} // namespace ru
